Tasks/AzureResourceGroupDeployment/models/DeployAzureRG.ts

```diff
@@ -106,14 +76,72 @@ export class AzureRGTaskParameters {
         }
     }
 
-    private getARMCredentials(connectedService: string): msRestAzure.ApplicationTokenCredentials {
+    private  async getARMCredentials(connectedService: string){
         var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
         var servicePrincipalKey: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
         var tenantId: string = tl.getEndpointAuthorizationParameter(connectedService, "tenantid", false);
         var armUrl: string = tl.getEndpointUrl(connectedService, true);
         var envAuthorityUrl: string = tl.getEndpointDataParameter(connectedService, 'environmentAuthorityUrl', true);
-        envAuthorityUrl = (envAuthorityUrl != null) ? envAuthorityUrl : "https://login.windows.net/";
-        var credentials = new msRestAzure.ApplicationTokenCredentials(servicePrincipalId, tenantId, servicePrincipalKey, armUrl, envAuthorityUrl);
+        var environment: string = tl.getEndpointDataParameter(connectedService, 'environment', true);
+        var activeDirectoryResourceId: string = tl.getEndpointDataParameter(connectedService, 'activeDirectoryServiceEndpointResourceId', true);
+        
+        if(environment != null && environment.toLowerCase() == azureStackEnvironment.toLowerCase()) {
+            if(!envAuthorityUrl || !activeDirectoryResourceId) {
```

Comment:
armUrl - It acts as base url to make any REST request to azure. 
activeDirectoryResourceId - Used to fetch access token as app in active directory get registered with this Url. In case of all national cloud armUrl and activeDirectoryResourceId urls are same. But in case of Azure Stack both url differs.